### PR TITLE
New command 'mirror' for mirroring the frame tree

### PR DESCRIPTION
**Pull request by t-wissmann**
This adapts the 'mirror' implementation from the PR #93 to the new
codebase. Additionally, it accepts a positional argument along which
axis the layout is to be mirrored.

On the implementation side, it uses type_traits and a new 'is_finite'
predicate to avoid boilerplate code when instantiating the Converter
class to the enum class type Horizontal/Vertical/Both. There is still a
bit to write, but much less compared to the Converter-implementation for
the ClientPlacement class.

In order to make Converter applicable to type traits, a new template
parameter to Converter "typename = void" was necessary.

### CODE REVIEW

File: doc/herbstluftwm.txt [hunk 1]
```diff
@@ -519,6 +519,10 @@ rotate::
     Rotates the layout on the focused tag counterclockwise by 90 degrees. This
     only manipulates the alignment of frames, not the content of them.
 
+mirror ['--vertical'|'--horizontal']::
```
Comment (blarz) [inline doc/herbstluftwm.txt:522]: I know that you want to use the new ArgParse functionality,
but does it make sense to have the layout mirror direction as optional flags?

A positional argument vertical/horizontal/both may also be a solution, right?
Either way is fine. But mirror without any flags is a no-op then, right?

Comment (blarz) [reply doc/herbstluftwm.txt:522]: But playing around with it right now, pretty nice functionality anyway!

Comment (t-wissmann) [reply doc/herbstluftwm.txt:522]: > A positional argument vertical/horizontal/both may also be a solution, right?

Right, I never thought about that. I think `vertical`/`horizontal`/`both` (with `horizontal` being the default) is more intuitive

**Comment by t-wissmann** [comment]
I've updated the `mirror` command to have positional arguments now. As a side effect, this PR is not independent of PR #1045.

**Comment by blarz** [comment]
> I've updated the `mirror` command to have positional arguments now. As a side effect, this PR is not independent of PR #1045.

Awesome I'll have a look at it tonight. Can you update the documentation, it still refers to the flags. Otherwise it looks good when I flew over the code.

**Comment by t-wissmann** [comment]
> Awesome I'll have a look at it tonight. Can you update the documentation, it still refers to the flags. Otherwise it looks good when I flew over the code.

I've updated the man page. Since the CI does not check the doc, I saved some cpu time and cancelled the travis CI build, so don't be worried by the CI failure for 6de9972. The ci for the previous commit of this PR (6a5e75d) succeeded.

**Comment by blarz** [comment]
> > Awesome I'll have a look at it tonight. Can you update the documentation, it still refers to the flags. Otherwise it looks good when I flew over the code.
> 
> I've updated the man page. Since the CI does not check the doc, I saved some cpu time and cancelled the travis CI build, so don't be worried by the CI failure for [6de9972](https://github.com/herbstluftwm/herbstluftwm/commit/6de9972060faa2746b72356b35bf0dc320ddcace). The ci for the previous commit of this PR ([6a5e75d](https://github.com/herbstluftwm/herbstluftwm/commit/6a5e75d10efb5fe79fd55f8f4fcffa81b010f7e6)) succeeded.

Thanks! It'll be build now anyway because of mergify. Let's see what happens :)

**Review by blarz — APPROVED** [review]
Nice!

**Pull request merged**